processor/processor.go

```diff
@@ -47,14 +48,21 @@ func (p *PayloadProcessor) Name() string {
 	return p.ProcessorName
 }
 
-func (p *PayloadProcessor) Decode(raw map[string]interface{}) (model.Payload, error) {
+func (p *PayloadProcessor) Decode(raw map[string]interface{}) (*metadata.Metadata, []transform.Eventable, error) {
```

Comment:
I agree of course. `ProcessorName` was just because it was conflicting with `Name()` which needed to be on the interface. I'll rethinking the structure to see if we can rid of the `Name()` completely.

Reply:
I ended up completely removing `DecodePayload` because those methods (`transaction.DecodePayload`, `error.DecodePayload`) were pretty much copies of each other and they could be abstracted into a single method: https://github.com/elastic/apm-server/pull/1183/files#diff-1d1d00824b3268cdd91ecb964e33cddaR52.